embassy-mspm0/src/i2c.rs

```diff
@@ -83,6 +70,21 @@ impl ClockDiv {
     }
 }
 
+impl From<ClockDiv> for vals::Ratio {
+    fn from(value: ClockDiv) -> Self {
+        match value {
+            ClockDiv::DivBy1 => Self::DIV_BY_1,
+            ClockDiv::DivBy2 => Self::DIV_BY_2,
+            ClockDiv::DivBy3 => Self::DIV_BY_3,
+            ClockDiv::DivBy4 => Self::DIV_BY_4,
+            ClockDiv::DivBy5 => Self::DIV_BY_5,
+            ClockDiv::DivBy6 => Self::DIV_BY_6,
+            ClockDiv::DivBy7 => Self::DIV_BY_7,
+            ClockDiv::DivBy8 => Self::DIV_BY_8,
+        }
+    }
+}
+
```

Comment:
I think we did this as an associated function rather than a trait impl to avoid making `From<ClockDiv> for vals::Ratio` public API. Probably should revert this part.